src/huggingface_hub/repository.py

```diff
@@ -160,32 +164,42 @@ def clone_from(self, repo_url: str, use_auth_token: Union[bool, str, None] = Non
         is_hf_id = len(repo_url.split("/")) <= 2
         api = HfApi()
 
+        if not is_hf_id and repo_url.split("/")[-3] == "datasets":
+            self.repo_type = "dataset"
+
         if token is not None:
+            user, valid_organisations = api.whoami(token)
             if is_hf_url:
-                organization, repo_id = repo_url.split("/")[-2:]
+                namespace, repo_id = repo_url.split("/")[-2:]
             elif is_hf_id:
                 if len(repo_url.split("/")) == 2:
                     # Passed <user>/<model_id> or <org>/<model_id>
-                    organization, repo_id = repo_url.split("/")[-2:]
+                    namespace, repo_id = repo_url.split("/")[-2:]
                 else:
                     # Passed <model_id>
-                    organization, repo_id = api.whoami(token)[0], repo_url
+                    namespace, repo_id = user, repo_url
+
+                repo_url = ENDPOINT
+
+                if self.repo_type == "dataset":
+                    repo_url += "/datasets"
 
-                repo_url = f"{ENDPOINT}/{organization}/{repo_id}"
+                repo_url += f"/{namespace}/{repo_id}"
             else:
                 raise ValueError(
                     f"Unable to retrieve user and repo ID from the passed clone URL: {repo_url}"
                 )
 
             repo_url = repo_url.replace("https://", f"https://user:{token}@")
 
-            print("Creating model under", repo_id, organization)
-            api.create_repo(
-                token,
-                repo_id,
-                organization=organization,
-                exist_ok=True,
-            )
+            if namespace == user or namespace in valid_organisations:
```

Comment:
are you sure that this validation is worth an extra call to `api.whoami`?

Reply:
Actually it's not only that validation leveraging the `api.whoami`, it's also when the user hasn't specified an ID or organization alongside their model ID.

Also, this validation is important! If we don't validate that here, then if I'm doing the following from my account:
```py
Repository("xxx", clone_from="julien-c/dummy-model")
```
with `dummy-model` being an actual model on your namespace, then it would error with the request to create it (`exist_ok` doesn't care if it already exists but if it's not in your namespace), while we would expect the clone to happen irrespective of your ability to push to it.

This check ensures this works correctly.